pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go

```diff
@@ -0,0 +1,400 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package assemble
+
+import (
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1alpha2"
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
+	ctrlutil "github.com/oam-dev/kubevela/pkg/controller/utils"
+	"github.com/oam-dev/kubevela/pkg/oam"
+	"github.com/oam-dev/kubevela/pkg/oam/util"
+)
+
+// NewAppManifests create a AppManifests
+func NewAppManifests(appRevision *v1beta1.ApplicationRevision) *AppManifests {
+	return &AppManifests{AppRevision: appRevision}
+}
+
+// AppManifests contains configuration to assemble resources recorded in the ApplicationRevision.
+// 'Assemble' means expand Application(Component and Trait) into K8s resource and get them ready to go, to be emitted
+// into K8s
+type AppManifests struct {
+	AppRevision     *v1beta1.ApplicationRevision
+	WorkloadOptions []WorkloadOption
+
+	appComponents []applicationComponent
+	components    []*v1alpha2.Component
```

Comment:
what's these two difference?

Reply:
appCompoents = appConfigComponent ([] {a comp + its traits + its scopes});
comp = raw workload + helm

Reply:
understood, I think we should use something new like `models` to record this thing, and the `models` is just a memory struct that won't exist in K8s

Reply:
Totally agreed!  These v1alpha2 APIs can retire now.